### PR TITLE
Add picross (nonogram) game

### PR DESCRIPTION
**Pull request by basic-experimental**
This is a picross puzzle player. It takes the puzzles in the form of [`.non` files](https://github.com/mikix/nonogram-db/blob/master/FORMAT.md). Currently, this is not ready to be merged, as it is missing many features and doesn't look great. I would appreciate feedback on the design direction and other features.
Currently, there is no level selector, so it will load `demo.non` as soon as the page is loaded. So far, it seems the only program on the website that has a selector like this would be the barcode generator. Should I try to keep it consistent with that, or do something new?
It is possible to make a level generator, but I find it way less fun when there isn't any theme to the completed puzzle. That does bring the question of where to get the puzzles. The [repo that the format information is in](https://github.com/mikix/nonogram-db) also contains puzzles with various licenses. This is where `demo.non` came from. Should I just use some of those? I could also create some myself.

In case you don't want to solve the puzzle yourself, [click here for the solution](https://github.com/Wolfyxon/3ds-web-stuff/assets/169532384/b0ddd1aa-f38e-445a-bd76-a44a9796cbe8)

### CODE REVIEW

**Comment by Trickiy** [comment]
Some ideas for the direction 1.) Should you be consistent with the barcode generator? I think that there should be some experimentation. As having all games having a samey look to it would make things bland looking and repetitive. A idea I have is having the menus be a pixlated grid. That would slide the direction of were the page is at, and have the text be a retro pixlated looking too. While having what is hovered over pop out a bit, do a little hop, and then rotate to the left and right a little. 2.) Where should the puzzles be obtained? Use some of what is in the repo as a starting base with your own (If I can find a way to make pixel art within my restricted browser. I will give some screenshots for ideas.) and add the generator as endgame content.

**Comment by magiczocker10** [comment]
@basic-experimental @Trickiy  Do you have discord for easier communication?

**Comment by Trickiy** [comment]
> @basic-experimental @Trickiy Do you have discord for easier communication?

I'm in a bad position at the moment sadly so no I do not. When I am in a better position I  will have discord, but it would be faster if brain dead Biden doesn't win the election. (Wich is very likely due to curruption in the gov at the moment I've heard.) As I'd be able to get a job sooner.

**Comment by basic-experimental** [comment]
> @basic-experimental @Trickiy Do you have discord for easier communication?

I do. I am "das904."

Sorry for the wait, I wasn't able to respond.

**Comment by Trickiy** [comment]
> @basic-experimental @Trickiy Do you have discord for easier communication?

I am happy to say that I managed to find a way to access discord within my overly restricted sandbox browser. So here's my username trickiidude.

**Comment by Trickiy** [comment]
Basic could you add the ability to navigate the menu with the d-pad and press a to enter the level.

**Comment by basic-experimental** [comment]
All right, I'm going to mark this as ready for review now. Only the tutorial levels have been made at the moment, but the game works, at least on new 3ds. I don't think that the CSS transform effects should make any difference to browsers without it, and I think the rest of the code is fairly compatible.

File: games/picross/picross.js [hunk 1]
```diff
@@ -0,0 +1,433 @@
+window.addEventListener('load', function() {
+	function requestText(url, callback){
```
Comment (Wolfyxon) [inline games/picross/picross.js:2]: `requestText()` is redundant. **scripts/lib/net.js** has `httpGet()`

File: games/picross/picross.js [hunk 2]
```diff
@@ -0,0 +1,433 @@
...
+		}
+	}
+
+	Element.prototype.makeLastChild = function () {
+		tempDiv = this.parentNode.appendNew("div");
+		this.parentNode.insertBefore(this, tempDiv);
+		this.parentNode.removeChild(tempDiv);
+	}
+
+	String.prototype.startsWith = function (str) { // This is a polyfill
```
Comment (Wolfyxon) [inline games/picross/picross.js:94]: Polyfills should be in **global.js**

**Review by Wolfyxon — CHANGES_REQUESTED** [review]

**Comment by basic-experimental** [comment]
Oh oops, I didn’t mean to do that. Sorry! I’ll get on with fixing those now.

**Comment by Wolfyxon** [comment]
Please replace `requestText()` with `net.httpGet()`. It's not the same name, and it got moved to a namespace.

**Comment by Wolfyxon** [comment]
Oh also, the text doesn't fully fit on the top screen.
image
(ignore the red line)

**Comment by basic-experimental** [comment]
Unfortunately I need to leave for something now and will be away for seven hours. I will try to fix those when I come back.

**Comment by Wolfyxon** [comment]
Alright that's fine

**Comment by basic-experimental** [comment]
I've finished with those changes now and tested them again. It all seems good. A real n3DS still has a tiny bit of text overflow on Tutorial-2, but it is still easily readable.

**Review by Wolfyxon — APPROVED** [review]
Looks good

**Pull request merged**